src/test/unit/mcmc/hmc/hamiltonians/diag_e_metric_test.cpp

```diff
@@ -12,48 +12,48 @@
 typedef boost::ecuyer1988 rng_t;
 
 TEST(McmcDiagEMetric, sample_p) {
-  
+
   rng_t base_rng(0);
-  
+
   Eigen::VectorXd q(2);
   q(0) = 5;
   q(1) = 1;
-  
+
   stan::mcmc::mock_model model(q.size());
   stan::mcmc::diag_e_metric<stan::mcmc::mock_model, rng_t> metric(model);
   stan::mcmc::diag_e_point z(q.size());
-  
+
   int n_samples = 1000;
   double m = 0;
   double m2 = 0;
-  
+
   for (int i = 0; i < n_samples; ++i) {
     metric.sample_p(z, base_rng);
-    double T = metric.T(z);
-    
-    double delta = T - m;
+    double tau = metric.tau(z);
+
+    double delta = tau - m;
     m += delta / static_cast<double>(i + 1);
```

Comment:
(fix whenever + Future) You don't need to static cast one of two arguments --- it's built into all the arithmetic operators.  So this can just be `delta / (i + 1)`.